File: app/src/main/java/ru/art2000/calculator/view/settings/InfoActivity.java
package ru.art2000.calculator.view.settings;

import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.widget.ImageButton;
import android.widget.Space;

import androidx.lifecycle.ViewModelProvider;

import java.util.Objects;

import ru.art2000.calculator.R;
import ru.art2000.calculator.databinding.ActivityAppInfoBinding;
import ru.art2000.calculator.databinding.AuthorLinkItemBinding;
import ru.art2000.calculator.model.settings.AuthorLink;
import ru.art2000.calculator.view_model.settings.InfoViewModel;
import ru.art2000.extensions.activities.AutoThemeActivity;

public class InfoActivity extends AutoThemeActivity {

    @Override
    public void onCreate(Bundle savedInstance) {
        super.onCreate(savedInstance);

        ActivityAppInfoBinding binding = ActivityAppInfoBinding.inflate(getLayoutInflater());

        setContentView(binding.getRoot());

        setSupportActionBar(binding.toolbar);
        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);

        InfoViewModel model = new ViewModelProvider(this).get(InfoViewModel.class);

        String changelogText = model.getChangeLogText();
        if (changelogText == null) changelogText = getString(R.string.changelog_load_failed);

        binding.changelog.setText(changelogText);

        int width;
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            width = getWindowManager().getCurrentWindowMetrics().getBounds().width();
        } else {
            DisplayMetrics dm = new DisplayMetrics();
            getWindowManager().getDefaultDisplay().getMetrics(dm);
            width = dm.widthPixels;
        }

        int imageSize = getResources().getDimensionPixelSize(R.dimen.author_info_link_image_size);

        int linksCount = model.getAuthorLinks().size();
        int gapSize = width / linksCount - imageSize;

        for (int i = 0; i < linksCount; ++i) {
            AuthorLink link = model.getAuthorLinks().get(i);

            ImageButton linkButton = AuthorLinkItemBinding.inflate(getLayoutInflater()).getRoot();

            linkButton.setImageResource(link.getImage());
            linkButton.setOnClickListener(v -> {
                String url = getString(link.getLink());
                Intent githubIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
                startActivity(githubIntent);
            });

            binding.linksBlock.addView(linkButton, imageSize, imageSize);

            if (i < linksCount - 1) {
                Space space = new Space(this);
                binding.linksBlock.addView(space, gapSize, 0);
            }

        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        finish();
        return true;
    }
}

File: settings/src/androidMain/kotlin/ru/art2000/calculator/settings/view/InfoActivity.kt
package ru.art2000.calculator.settings.view

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.Space
import androidx.activity.viewModels
import by.kirich1409.viewbindingdelegate.CreateMethod
import by.kirich1409.viewbindingdelegate.viewBinding
import ru.art2000.calculator.common.view.AppActivity
import ru.art2000.calculator.settings.R
import ru.art2000.calculator.settings.databinding.ActivityAppInfoBinding
import ru.art2000.calculator.settings.databinding.AuthorLinkItemBinding
import ru.art2000.calculator.settings.model.AuthorLink
import ru.art2000.calculator.settings.vm.InfoViewModel
import ru.art2000.extensions.views.isLandscape
import androidx.core.net.toUri

internal class InfoActivity : AppActivity() {

    private val model by viewModels<InfoViewModel>()
    private val binding by viewBinding<ActivityAppInfoBinding>(CreateMethod.INFLATE)

    override val topViews: List<View>
        get() = listOf(binding.appBarLayout)

    override val bottomViews: List<View>
        get() = when (isLandscape) {
            true -> listOf(binding.changelogScrollview, binding.linksBlock)
            false -> listOf(binding.root)
        }

    override val leftViews: List<View>
        get() = listOf(binding.root)

    override val rightViews: List<View>
        get() = listOf(binding.root)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(binding.root)
        setSupportActionBar(binding.toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        binding.changelog.text = model.changeLogText ?: getString(R.string.changelog_load_failed)

        binding.linksBlock.post {
            addLinks(binding.linksBlock)
        }
    }

    private fun addLinks(linksRoot: ViewGroup) {
        val width = linksRoot.width
        val imageSize = resources.getDimensionPixelSize(R.dimen.author_info_link_image_size)
        val linksCount = model.authorLinks.size
        val gapSize = width / linksCount - imageSize
        repeat(linksCount) { i ->
            val link = model.authorLinks[i]
            val linkButton = AuthorLinkItemBinding.inflate(layoutInflater).root
            linkButton.setImageResource(link.image)
            linkButton.setOnClickListener {
                onLinkClick(link)
            }
            linksRoot.addView(linkButton, imageSize, imageSize)
            if (i < linksCount - 1) {
                val space = Space(this)
                linksRoot.addView(space, gapSize, 0)
            }
        }
    }

    private fun onLinkClick(link: AuthorLink) {
        val url = getString(link.link)
        val uriIntent = Intent(Intent.ACTION_VIEW, url.toUri())
        startActivity(uriIntent)
    }

    override fun onSupportNavigateUp(): Boolean {
        finish()
        return true
    }
}